src/transformer.js

```diff
@@ -1,58 +1,53 @@
-import { basename } from "path";
-import { pathToFileURL } from "url";
-import svelte from "svelte/compiler";
+import { basename } from 'path'
+import { pathToFileURL } from 'url'
+import * as svelte from 'svelte/compiler'
 
-import { getSvelteConfig } from "./svelteconfig";
+import { getSvelteConfig } from './svelteconfig.cjs'
 
-const dynamicImport = async (filename) => import(pathToFileURL(filename));
+const dynamicImport = async (filename) => import(pathToFileURL(filename))
 
-const transformer =
-  (options = {}) =>
-  async (source, filename) => {
-    const { preprocess, rootMode } = options;
-    if (!preprocess) {
-      return compiler(options, filename, source);
-    }
+export const processAsync = async (source, filename, jestOptions) => {
+  const options = jestOptions?.transformerConfig ?? {}
+  const { preprocess, rootMode } = options
 
-    const svelteConfigPath = getSvelteConfig(rootMode, filename, preprocess);
-    const svelteConfig = await dynamicImport(svelteConfigPath);
-    const processed = await svelte.preprocess(
-      source,
-      svelteConfig.default.preprocess || {},
-      { filename }
-    );
+  if (!preprocess) {
+    return compiler(options, filename, source)
+  }
+
+  const svelteConfigPath = getSvelteConfig(rootMode, filename, preprocess)
+  const svelteConfig = await dynamicImport(svelteConfigPath)
+  const processed = await svelte.preprocess(
+    source,
+    svelteConfig.default.preprocess || {},
+    { filename }
+  )
 
-    return compiler(options, filename, processed.code, processed.map);
-  };
+  return compiler(options, filename, processed.code, processed.map)
+}
 
 const compiler = (options = {}, filename, processedCode, processedMap) => {
-  const { debug, compilerOptions } = options;
+  const { debug, compilerOptions } = options
 
   const result = svelte.compile(processedCode, {
     filename: basename(filename),
     css: true,
     accessors: true,
     dev: true,
-    format: "esm",
+    format: 'esm',
     sourcemap: processedMap,
-    ...compilerOptions,
-  });
+    ...compilerOptions
+  })
 
   if (debug) {
-    console.log(result.js.code);
+    console.log(result.js.code)
   }
 
-  const esInterop =
-    'Object.defineProperty(exports, "__esModule", { value: true });';
-
   return {
-    code: result.js.code + esInterop,
-    map: JSON.stringify(result.js.map),
-  };
-};
-
-const createTransformer = (options) => ({
-  processAsync: transformer(options),
-});
+    code: result.js.code,
+    map: JSON.stringify(result.js.map)
+  }
+}
 
-export default createTransformer;
+export default {
+  processAsync
+}
```

Comment:
Needs to probably change to `export default processAsync` @sebastianrothe? I haven't read the new transformer docs so I'll touch back on this when I have time unless it's solved earlier.

Reply:
I have only found these references:
- https://github.com/sodatea/vite-jest/blob/main/packages/vite-jest/index.js
- https://github.com/axe-me/vite-jest/blob/main/src/transformer.ts
- https://github.com/bitttttten/jest-transformer-mdx/blob/da6eae4b9fe089643636092eeed7dc5f1a370673/mdx-options.cjs

Most of the transformers (like vue-jest, esbuild-jest) still use the old syntax and/or no ESM.

Reply:
The thing is, there is no documentation about asyncTransformers :( https://jestjs.io/docs/code-transformation#writing-custom-transformers